### ISSUE TITLE
Request 2 times

### ISSUE THREAD

**Issue by StringBOX** [danikula/AndroidVideoCache#181]
Using the Fiddler test, I found the network requested 2 times, didn't I just request 1 times?

**Comment by yyyzp** [comment]
maybe 3 or 4 times.how to fix it?

**Comment by Labmem003** [comment]
It request 2 times because it use 1 time to fetch the content-length. The read of video data actually happen in the second time。Since content-length is in header, you can  optimize the code by use head method instead of get method to squash the consume of the first time .IMO.

**Comment by m7mdra** [comment]
i agree with @Labmem003 , i built a small proxy server to mimic this library and i found out that the first request is HEAD to read the headers (including content-length) and the second request to read the actual data
